src/scripting/lua_widget_attributes.cpp

```diff
@@ -342,16 +420,328 @@ WIDGET_SETTER("marked_up_text", t_string, gui2::styled_widget)
 	w.set_label(value);
 }
 
+WIDGET_GETTER("characters_per_line", int, gui2::label)
+{
+	return w.get_characters_per_line();
+}
+
+WIDGET_SETTER("characters_per_line", int, gui2::label)
+{
+	if(value < 0) {
+		throw std::invalid_argument("characters_per_line must be >= 0");
+	}
+	w.set_characters_per_line(value);
+	INVALIDATE_LAYOUT;
+}
+
+WIDGET_GETTER("editable", bool, gui2::text_box)
+{
+	return w.is_editable();
+}
+
+WIDGET_SETTER("editable", bool, gui2::text_box)
+{
+	w.set_editable(value);
+}
+
+WIDGET_GETTER("ellipse_mode", std::string, gui2::styled_widget)
```

Comment:
This is the wrong name – "ellipse" refers to a squashed circle, not to an omission of words. It should be "ellipsis" or "ellipsize" – probably the latter since it matches up with what Pango calls it.

Reply:
I thought that was odd, but internally it's called ellipse so I figured there was a reason and just went with it.

Reply:
Sounds like someone else made the mistake and you just went with it… but we shouldn't propagate that mistake into the Lua API.